ui/src/store/store.ts

```diff
@@ -1,104 +1,90 @@
 import { configureStore } from '@reduxjs/toolkit';
-import { trackLayoutReducer } from 'track-layout/track-layout-store';
-import { infraModelReducer } from 'infra-model/infra-model-store';
+import { trackLayoutReducer } from 'track-layout/track-layout-slice';
+import { infraModelReducer } from 'infra-model/infra-model-slice';
+import { dataProductsReducer } from 'data-products/data-products-slice';
+import { persistReducer } from 'redux-persist';
 import storage from 'redux-persist/lib/storage';
 import { combineReducers } from 'redux';
-import { persistReducer } from 'redux-persist';
+
 import thunk from 'redux-thunk';
-import { dataProductsReducer } from 'data-products/data-products-store';
+import { commonReducer } from 'common/common-slice';
 
 export const RESET_STORE_ACTION = {
     type: 'RESET_STORE',
 };
 
-const trackLayoutReducers = combineReducers({
-    trackLayout: trackLayoutReducer,
-});
-
-const inframodelReducers = combineReducers({
-    infraModel: infraModelReducer,
-});
-
-const dataProductsReducers = combineReducers({
-    dataProducts: dataProductsReducer,
-});
-
-// Wrap combined reducers to handle root level state
-const trackLayoutRootReducer: typeof trackLayoutReducers = (state, action) => {
-    if (action.type == RESET_STORE_ACTION.type) {
-        // Reset to initial state
-        return trackLayoutReducers(undefined, action);
-    }
-    return trackLayoutReducers(state, action);
-};
-
-const inframodelRootReducer: typeof inframodelReducers = (state, action) => {
-    if (action.type == RESET_STORE_ACTION.type) {
-        // Reset to initial state
-        return inframodelReducers(undefined, action);
-    }
-    return inframodelReducers(state, action);
+const trackLayoutPersistConfig = {
+    key: 'rootTracklayout',
+    storage,
 };
 
-const dataProductsRootReducer: typeof dataProductsReducers = (state, action) => {
+const trackLayoutAppLevelReducer: typeof trackLayoutReducer = (state, action) => {
```

Comment:
Nämä slice-kohtaiset muuttujat olisi ollut kiva viedä omiin sliceihinsä, mutta niiden siirtäminen sai aikaan virheen ("TypeError: Basereducer is not a function).